examples/vision/adain.py

```diff
@@ -0,0 +1,675 @@
+
+test_style, test_content = next(iter(test_ds))
+
+
+class TrainMonitor(tf.keras.callbacks.Callback):
+    def on_epoch_end(self, epoch, logs=None):
+        # Encode the style and content image.
+        test_style_enc = self.model.encoder(test_style)
+        test_content_enc = self.model.encoder(test_content)
+
+        # Compute the AdaIN features.
+        test_t = self.model.ada_in(style=test_style_enc, content=test_content_enc)
+        test_recons_image = self.model.decoder(test_t)
+
+        # Plot the Style, Content and the NST image.
+        fig, ax = plt.subplots(nrows=1, ncols=3, figsize=(20, 5))
+        ax[0].imshow(tf.keras.preprocessing.image.array_to_img(test_style[0]))
+        ax[0].set_title(f"Style: {epoch:03d}")
+
+        ax[1].imshow(tf.keras.preprocessing.image.array_to_img(test_content[0]))
+        ax[1].set_title(f"Content: {epoch:03d}")
+
+        ax[2].imshow(tf.keras.preprocessing.image.array_to_img(test_recons_image[0]))
+        ax[2].set_title(f"NST: {epoch:03d}")
+
+        plt.show()
+        plt.close()
+
+
+"""
+# Train the model
+
+In this section, we initialize the optimizer, loss funtion and the
+trainer module. We compile the trainer module with the optimizer and
+the loss function and then train it.
+
+*Note*: We train the model for a single epoch for time constranints,
+but we will need to train is for atleast 30 epochs to see good results.
+"""
+
+optimizer = keras.optimizers.Adam(learning_rate=1e-5)
+loss_fn = keras.losses.MeanSquaredError()
+
+encoder = get_encoder()
+loss_net = get_loss_net()
+decoder = get_decoder()
+
+model = NST(
+    encoder=encoder, decoder=decoder, ada_in=ada_in, loss_net=loss_net, lamb=4.0
+)
+
+model.compile(optimizer=optimizer, loss_fn=loss_fn)
+
+history = model.fit(
+    train_ds,
+    epochs=EPOCHS,
+    steps_per_epoch=50,
+    validation_data=val_ds,
+    validation_steps=50,
+    callbacks=[TrainMonitor()],
+)
+
+"""
+# Inference
+
+After we train the model, we now need to run inference on it. We will
+pass arbitrary content and style images from the test dataset and see
+the NST images.
+
+*NOTE*: To try out the model on your own images you can go to this
+[Hugging Face demo](https://huggingface.co/spaces/ariG23498/nst).
+"""
+
+for style, content in test_ds.take(1):
+    style_enc = model.encoder(style)
+    content_enc = model.encoder(content)
+    t = model.ada_in(style=style_enc, content=content_enc)
+    recons_image = model.decoder(t)
+    fig, axes = plt.subplots(nrows=10, ncols=3, figsize=(10, 30))
+    [ax.axis("off") for ax in np.ravel(axes)]
+
+    for (ax_s, ax_c, ax_r), s_im, c_im, r_im in zip(
+        axes, style[0:10], content[0:10], recons_image[0:10]
+    ):
+        ax_s.imshow(s_im)
+        ax_s.set_title("Style Image")
+        ax_c.imshow(c_im)
+        ax_c.set_title("Content Image")
+        ax_r.imshow(r_im)
+        ax_r.set_title("NST Image")
+
+"""
+# Conclusion
+
+Adaptive Instance Normalization is an important step in the direction
```

Comment:
we can remove "is an important step in the direction of Neural Style Transfer research because it" resulting in...

`Adaptive Instance Normalization allows arbitrary style
transfer in real time`